MyWeight/Extensions/UITableViewCell+ReuseIdentifier.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  UITableViewCell+ReuseIdentifier.swift
+//  MyWeight
+//
+//  Created by Bruno Koga on 22/06/16.
+//  Copyright © 2016 Diogo Tridapalli. All rights reserved.
+//
+
+import UIKit
+
+extension UITableViewCell {
+    /**
+     - returns: a UINib with the default Nib name (matches the class name).
+        If the nib file is not found, this will cause a exception crash.
+     */
+    static func nib() -> UINib {
+        return UINib(nibName: defaultNibName, bundle: nil)
+    }
+    
+    /**
+     Default nib name for this class (matches the class name).
+     */
+    static var defaultNibName: String {
+        return String(self)
+    }
+    
+    /**
+     Default reuse identifier for this class (matches the class name).
+     */
+    static var defaultReuseIdentifier: String {
+        return String(self)
+    }
+}
+
+extension UITableView {
+    
+    /**
+     Register a cell into the tableView.
+     Uses the defaultReuseIdentifier if the reuseIdentifier is not provided.
+     
+     - parameter cellClass: the class to register.
+     */
+    func registerCellNib(cellClass: UITableViewCell.Type, forCellReuseIdentifier reuseIdentifier: String? = nil) {
```

Comment:
Ok, so I reverted the last changes. Here's why:

You proposed change broke the tests. Basically, this proposed change:

```swift
func registerCellClass<CellType where CellType: UITableViewCell>(cellClass: CellType.Type, forCellReuseIdentifier reuseIdentifier: String = CellType.defaultReuseIdentifier) {
    self.registerClass(cellClass.self, forCellReuseIdentifier: reuseIdentifier)
}
```

treats the CellType like a member of UITableViewCell class. So, no matters what subclass you pass to this func, the CellType.Type would be `UITableViewCell` and the defaultReuseIdentifier would also be `UITableViewCell`.

Don't know if this is a swift bug or the intended behavior (or if I did something wrong). If you have time to check this, please go ahead and verify/fix. Otherwise I'd suggest us to merge this (since it's not a big deal anyway).